src/middlewares/auth.middleware.ts

```diff
@@ -0,0 +1,56 @@
+import {Request, Response, NextFunction} from "express";
+import {JWT} from "@/config";
+import jwt from "jsonwebtoken";
+
+declare module "express-serve-static-core" {
+    interface Request {
+        user?: {id: number, role: string, email?: string}
+    }
+}
+
+export async function authMiddleware(
+    req: Request, 
+    res:Response, 
+    next: NextFunction){
+
+        try{
+            const authHeader = req.headers["authorization"];
+
+            if (!authHeader || !authHeader.startsWith("Bearer ")){
```

Comment:
hope you understand what this bearer means and how it'll dictate how you pass in the token during requests